### PR TITLE
Add range to iter

### LINKED ISSUE #537: add Iter::range utils
see this https://github.com/c-cube/iter?tab=readme-ov-file for inspirations

### PR DESCRIPTION
**Pull request by qazxcdswe123**
close #537

### CODE REVIEW

File: builtin/iter.mbt [hunk 1]
```diff
@@ -152,6 +152,30 @@ pub fn Iter::repeat[T](a : T) -> Iter[T] {
...
+/// # Arguments
+/// 
+/// * `start` - The starting value of the range (inclusive).
+/// * `end` - The ending value of the range (exclusive).
+/// 
+/// # Returns
+/// 
+/// Returns an iterator that iterates over the range of integers from `start` to `end - 1`.
+pub fn Iter::range(start : Int, end : Int) -> Iter[Int] {
+  Iter(
```
Comment (bobzhang) [inline builtin/iter.mbt:166]: 1. ~step : Int = 1
2. range_int64
3. range_double or better names?

File: builtin/builtin.mbti [hunk 2]
```diff
@@ -137,6 +137,9 @@ impl Iter {
   map[T, R](Self[T], (T) -> R) -> Self[R]
   op_add[T](Self[T], Self[T]) -> Self[T]
   prepend[T](Self[T], T) -> Self[T]
+  range(Int, Int, ~step : Int = ..) -> Self[Int]
+  range_Double(Double, Double, ~step : Double = ..) -> Self[Double]
```
Comment (bobzhang) [inline builtin/builtin.mbti:141]: T::upto/T::downto

**Comment by qazxcdswe123** [comment]
Not sure about the inclusive on both side approach.

File: builtin/iter_test.mbt [hunk 3]
```diff
@@ -242,6 +242,52 @@ test "collect" {
...
+    content="[0.0, 1.0, 2.0, 3.0, 4.0, 0.0, 1.0, 2.0]",
+  )?
+  inspect(
+    (-50).to(50, step=10).take(100).collect(),
+    content="[-50, -40, -30, -20, -10, 0, 10, 20, 30, 40, 50]",
+  )?
+  inspect(
+    0.0.to(1.0, step=0.1).take(5).collect(),
+    content="[0.0, 0.1, 0.2, 0.30000000000000004, 0.4]",
+  )?
```
Comment (bobzhang) [inline builtin/iter_test.mbt:274]: can you try 0.0.to(0.3) and see what will happen?

File: builtin/iter_test.mbt [hunk 4]
```diff
@@ -242,6 +242,52 @@ test "collect" {
...
+  inspect(
+    0x80000000L.to(0x80000000L + 10L, step=2L).take(5).collect(),
+    content="[2147483648, 2147483650, 2147483652, 2147483654, 2147483656]",
```
Comment (bobzhang) [inline builtin/iter_test.mbt:277]: let's make it `[a,b)`

Comment (qazxcdswe123) [reply builtin/iter_test.mbt:277]: My concern is, 0 to 10 sounds like `0-10`, and this is also what ruby does.
But exclusive end should be better for programming though.
Screenshot 2024-06-17 at 16 51 47

File: builtin/iter_test.mbt (hunk 3 above)
Comment (qazxcdswe123) [reply builtin/iter_test.mbt:275]: it would be `[0.0, 0.1, 0.2, 0.30000000000000004]`, as intended

**Pull request merged**